src/js/firstDiscoveryEditor.js

```diff
@@ -37,11 +35,78 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
                         }
                     }
                 }
+            },
+            navButtons: {
+                type: "gpii.firstDiscovery.navButtons",
+                container: "{that}.dom.navButtons",
+                createOnEvent: "onCreateNavButtons",
+                options: {
+                    gradeNames: ["fluid.prefs.msgLookup"],
+                    members: {
+                        messageResolver: "{firstDiscoveryEditor}.msgResolver"
+                    },
+                    strings: {
+                        back: "{that}.msgLookup.back",
+                        next: "{that}.msgLookup.next",
+                        start: "{that}.msgLookup.start",
+                        finish: "{that}.msgLookup.finish"
+                    },
+                    styles: "{firstDiscoveryEditor}.options.styles",
+                    panelTotalNum: "{firstDiscoveryEditor}.panels.length",
+                    model: {
+                        currentPanelNum: "{firstDiscoveryEditor}.model.currentPanelNum"
+                    }
+                }
+            }
+        },
+        selectors: {
+            prefsEditor: ".gpiic-fd-prefsEditor",
+            panel: ".gpiic-fd-prefsEditor-panel",
+            navButtons: ".gpiic-fd-navButtons"
+        },
+        styles: {
+            show: "gpii-fd-show",
+            currentPanel: "gpii-fd-current"
+        },
+        model: {
+            currentPanelNum: 1
+        },
+        modelListeners: {
+            "currentPanelNum": {
+                listener: "{that}.showPanel",
+                excludeSource: "init"
             }
         },
         events: {
-            onPrefsEditorReady: null
+            onPrefsEditorReady: null,
+            onCreateNavButtons: null
+        },
+        listeners: {
+            "onPrefsEditorReady.setPanels": {
+                listener: "gpii.firstDiscovery.setPanels",
+                args: ["{that}"],
+                priority: "first"
+            },
+            "onPrefsEditorReady.showInitialPanel": "{that}.showPanel",
+            "onPrefsEditorReady.createNavButtons": {
+                listener: "{that}.events.onCreateNavButtons"
+            }
+        },
+        invokers: {
+            showPanel: {
+                funcName: "gpii.firstDiscovery.showPanel",
+                args: ["{that}.panels", "{that}.model.currentPanelNum", "{that}.options.styles.currentPanel"]
+            }
         }
     });
 
+    gpii.firstDiscovery.setPanels = function (that) {
+        that.panels = that.prefsEditor.locate("panel");
```

Comment:
I've improved the code to be declarative as a "onPrefsEditorReady" listener. 

This pull request is ready for more review.